rllib/evaluation/worker_set.py

```diff
@@ -139,6 +140,17 @@ def __init__(
                     or not trainer_config.get("action_space")
                 )
             ):
+                # First assert that the first worker is ready.
+                # Not doing so leads to worker-not-ready errors sometimes on Windows.
+                for _ in range(10):
+                    try:
+                        test = ray.get(self.remote_workers()[0].apply.remote(lambda w: w.worker_index))
+                        assert test == 1
+                        break
+                    except Exception as e:
```

Comment:
I got curious and asked about what's the recommended way of doing this in #open-source.
I wonder why you need to have this try/except/sleep loop, since the folks are telling me ray.get(worker.func()) should tell you that the worker is ready, and block otherwise.

Reply:
I completely agree. Thanks for asking there. I was also thinking (after I did the PR :) ), we should maybe handle this more gracefully. It was never an issue before, but maybe something Windows-specific (the A3C/IMPALA tests reliably fail on Win).

Reply:
ok, can you please add some comments about the try catch block here, and we can call it a day?
thanks for investigating this man!!